opencti-platform/opencti-front/src/components/AppThemeProvider.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FunctionComponent, useContext } from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { createTheme, StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
+import { ThemeOptions } from '@mui/material/styles/createTheme';
+import { UserContext, UserContextType } from '../utils/Security';
+import themeDark from './ThemeDark';
+import themeLight from './ThemeLight';
+import {
+  useDocumentFaviconModifier,
+  useDocumentModifier,
+  useDocumentThemeModifier,
+} from '../utils/hooks/useDocumentModifier';
+import { AppThemeProvider_settings$data } from './__generated__/AppThemeProvider_settings.graphql';
+
+interface AppThemeProviderProps {
+  children: React.ReactNode
+  settings: AppThemeProvider_settings$data
+}
+
+const themeBuilder = (settings: AppThemeProvider_settings$data, themeColor: string) => {
+  if (themeColor === 'light') {
+    const platformThemeLightLogo = settings?.platform_theme_light_logo;
+    const platformThemeLightBackground = settings?.platform_theme_light_background;
+    const platformThemeLightPaper = settings?.platform_theme_light_paper;
+    const platformThemeLightNav = settings?.platform_theme_light_nav;
+    const platformThemeLightPrimary = settings?.platform_theme_light_primary;
+    const platformThemeLightSecondary = settings?.platform_theme_light_secondary;
+    const platformThemeLightAccent = settings?.platform_theme_light_accent;
+    return themeLight(
+      platformThemeLightLogo,
+      platformThemeLightBackground,
+      platformThemeLightPaper,
+      platformThemeLightNav,
+      platformThemeLightPrimary,
+      platformThemeLightSecondary,
+      platformThemeLightAccent,
+    );
+  }
+  const platformThemeDarkLogo = settings?.platform_theme_dark_logo ?? null;
+  const platformThemeDarkBackground = settings?.platform_theme_dark_background ?? null;
+  const platformThemeDarkPaper = settings?.platform_theme_dark_paper ?? null;
+  const platformThemeDarkNav = settings?.platform_theme_dark_nav ?? null;
+  const platformThemeDarkPrimary = settings?.platform_theme_dark_primary ?? null;
+  const platformThemeDarkSecondary = settings?.platform_theme_dark_secondary ?? null;
+  const platformThemeDarkAccent = settings?.platform_theme_dark_accent ?? null;
+  return themeDark(
+    platformThemeDarkLogo,
+    platformThemeDarkBackground,
+    platformThemeDarkPaper,
+    platformThemeDarkNav,
+    platformThemeDarkPrimary,
+    platformThemeDarkSecondary,
+    platformThemeDarkAccent,
+  );
+};
+
+const AppThemeProvider: FunctionComponent<AppThemeProviderProps> = (props) => {
+  const { children } = props;
```

Comment:
Would be better to destructure in the function call to avoid "props"

Reply:
Yes done.